### ISSUE TITLE
XML attributes are forced to be ordered

### ISSUE THREAD

**Issue by RicArch97** [labwc/labwc#2391]
### Problem description

I wrote a stylesheet using XSLT for converting from the Nix language to XML to autogenerate the rc.xml from Nix.

However i ran into an ordering problem when defining keybinds:
```
00:00:00.000 [../src/config/rcxml.c:559] expect <action name=""> element first. nodename: 'command.action' content: '/nix/store/nkjjc70cidfc3wbpas7kqr7agasma59c-rofi-1.7.5+wayland3/bin/rofi -show drun -config $XDG_CONFIG_HOME/rofi/menu.rasi'
00:00:00.000 [../src/action.c:591] Missing required argument for Execute: command
00:00:00.000 [../src/config/rcxml.c:1759] Removed invalid keybind action
```

Note that the XML specification (https://www.w3.org/TR/REC-xml/#sec-starttags) states:
> Note that the order of attribute specifications in a start-tag or empty-element tag is not significant.

Which kind of makes sense, they are named attributes, why would order matter?

Edit: it doesn't seem to be the case for every setting. For example it tried it with margin:
```
<margin bottom="10" left="10" right="10" top="10"/>
```
It's a different order than how it's defined in the man page, but I don't see any errors because of it. So it's somewhat inconsistent across the options.

### Steps to reproduce

1. Define a keybind where the attributes are not in the same order as they appear in the man page.

### labwc build source

Release

### labwc version

0.8.1

### labwc environment

From a TTY or some display manager like lightdm

### Distribution

NixOS

**Comment by Consolatis** [comment]
Yes, actions in specific require the `name` attribute to come first. I am also not that happy about it and we should figure out how to make the argument ordering irrelevant.

The main issue with actions in specific is that we do create the action struct with the name argument. Maybe we could have a generic `action_create(void)` and provide some additional `action_set_name(const char *name)` instead. Or make `action_arg_from_xml_node()` also handle the name (+ potentially rename the function). It might require some more refactoring though because that function also does argument verification based on the action type (which in turn is set by the `name` tag / child-node). Or we need a temporary struct which holds all the XML tags / child-nodes before creating the action struct.
